### PR TITLE
[filesystem] Fixed file dialog opening folder

### LINKED ISSUE #4862: [filesystem] File dialog seems to use wrong folder when opened
### Description
[When a file dialog is opened, it's possible to pass a folder parameter](https://github.com/theia-ide/theia/blob/master/packages/filesystem/src/browser/file-dialog/file-dialog-service.ts#L44). What I would personally expect would be to see the file dialog getting opened in that folder, what instead happens is that [the parent of the folder parameter is used](https://github.com/theia-ide/theia/blob/master/packages/filesystem/src/browser/file-dialog/file-dialog-service.ts#L77).

This is not the behaviour I would expect and makes it impossible to open a folder that has no subfolders. 

Is there a reason behind this?

### PR DESCRIPTION
**Pull request by federicobozzini**
A file dialog is opened in the folder used as a parameter.

Fixes #4862

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
Sorry, I see you already updated the Electron version as well 😄

File: packages/filesystem/src/browser/file-dialog/file-dialog-service.ts [hunk 1]
```diff
@@ -71,10 +71,10 @@ export class DefaultFileDialogService {
         return undefined;
     }
 
-    protected async getRootNode(folderToOpen?: FileStat): Promise<DirNode | undefined> {
+    protected async getDirNode(folderToOpen?: FileStat): Promise<DirNode | undefined> {
         const folder = folderToOpen || await this.fileSystem.getCurrentUserHome();
         if (folder) {
-            const rootUri = new URI(folder.uri).parent;
```
Comment (akosyakov) [inline packages/filesystem/src/browser/file-dialog/file-dialog-service.ts]: we should still use a parent if someone passes `FileStat` with `false` as `isDirectory`

File: packages/filesystem/src/browser/file-dialog/file-dialog-service.ts [hunk 2]
```diff
@@ -71,10 +71,10 @@ export class DefaultFileDialogService {
         return undefined;
     }
 
-    protected async getRootNode(folderToOpen?: FileStat): Promise<DirNode | undefined> {
```
Comment (akosyakov) [inline packages/filesystem/src/browser/file-dialog/file-dialog-service.ts]: let's keep `getRootNode` to avoid breaking changes

**Comment by akosyakov** [comment]
`Open File` menu action should still show the parent directory.

**Comment by federicobozzini** [comment]
> `Open File` menu action should still show the parent directory. We should fix this client, also check whether it makes sense for other existing clients.

Should we? The current behaviour seems different than what I would expect.

At the moment if I click `Open File` instead of the getting the dialog open in the workspace root, I see it gets opened in the parent folder.

File: packages/filesystem/src/browser/file-dialog/file-dialog-service.ts (hunk 2 above)
Comment (federicobozzini) [reply packages/filesystem/src/browser/file-dialog/file-dialog-service.ts]: Done.

File: packages/filesystem/src/browser/file-dialog/file-dialog-service.ts (hunk 1 above)
Comment (federicobozzini) [reply packages/filesystem/src/browser/file-dialog/file-dialog-service.ts]: You are right, but passing a file reference using a parameter named `folder` would be quite naughty!

**Comment by akosyakov** [comment]
As far as I know it was done on purpose: https://github.com/theia-ide/theia/pull/332  cc @svenefftinge 

@federicobozzini please sign your commits and ECA: https://accounts.eclipse.org/users/akosyakov/eca

File: packages/filesystem/src/browser/file-dialog/file-dialog-service.ts (hunk 1 above)
Comment (akosyakov) [reply packages/filesystem/src/browser/file-dialog/file-dialog-service.ts]: who knows what devs will do 😄

**Comment by svenefftinge** [comment]
> Should we? The current behaviour seems different than what I would expect.

I did it like that because you probably want to switch to a sibling workspace as this is how many devs organize multiple workspaces locally.

**Comment by federicobozzini** [comment]
Is this PR accepted? Can it be merged?

**Pull request merged**